imports/plugins/core/orders/client/templates/workflow/shippingInvoice.js

```diff
@@ -58,14 +59,33 @@ Template.coreOrderShippingInvoice.events({
 
     const state = instance.state;
     const order = state.get("order");
+    const orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
     const discount = state.get("field-discount") || 0;
 
-    Meteor.call("orders/approvePayment", order, discount, (error) => {
-      if (error) {
-        // Show error
-        Logger.warn(error);
-      }
-    });
+    if (discount > orderTotal) {
+      Alerts.inline("Discount cannot be greater than original total price", "error", {
+        placement: "coreOrderShippingInvoice",
+        i18nKey: "order.invalidDiscount",
+        autoHide: 10000
+      });
+    } else {
+      Meteor.call("orders/approvePayment", order, discount, (error) => {
+        if (error) {
+          // Log & show error if client-side fails
```

Comment:
I did mean client-side. The thought process on this is `if (discount >orderTotal) {` is in place for client-side actions, i.e. clicking the "Apply Discount" button. If someone is able to bypass this by not using the UI and directly calling `Meteor.call("orders/approvePayment", order, discount, (error) => {` we still want something to explain what's happening. This should never happen if anyone is using platform in a traditional way however, so maybe we do want to remove the log?